### PR TITLE
HOTFIX: Add file upload support to the audit log

### PR DESCRIPTION
**Pull request by timothybest-usds**
## Related Issue or Background Info

Attempting a bulk upload and getting a 400 due to the audit log not being able to convert `Part` to JSON

Looking for a fix in so we can get the upload done by EOD tomorrow

### CODE REVIEW

File: backend/src/main/java/gov/cdc/usds/simplereport/db/model/auxiliary/RequestVariablesSerializer.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import java.util.stream.Collectors;
+import javax.servlet.http.Part;
+
+public class RequestVariablesSerializer extends JsonSerializer<Map<String, Object>> {
+
+  private Object parseValue(Object value) {
+    if (value instanceof Part) {
+      Part part = (Part) value;
+      try {
+        return new BufferedReader(
```
Comment (benwarfield-usds) [inline backend/src/main/java/gov/cdc/usds/simplereport/db/model/auxiliary/RequestVariablesSerializer.java:22]: I would like to think we don't need to do a line-by-line read and then rejoin into a bigger string again—granted that Java readers and streams are annoying, it seems like we should be able to read the whole thing in at one go. Then again, we could just leave it as a `List<String>` for serialization...

**Review by benwarfield-usds — APPROVED** [review]
There are a bunch of possible wrinkles and tweaks around the edges we could do here, but the basic question is "do we seriously need to save the entire upload body?" If we do, this seems like a reasonable approach (I stress tested it on my local build, and it held up until tomcat's safety switches cut me off) and probably generates a little bit of a laundry-list of additional safety-net code we should be thinking about to prevent this from exploding, if we are ever to let org-admin users use it instead of site-management users. If not, we could just ... skip the InputStream attribute of the `Part` and be OK.

@jeskew-gov, @jdorothy , what say ye?

File: backend/src/main/java/gov/cdc/usds/simplereport/db/model/auxiliary/RequestVariablesSerializer.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    return value;
+  }
+
+  @Override
+  public void serialize(Map<String, Object> value, JsonGenerator gen, SerializerProvider arg2)
+      throws IOException, JsonProcessingException {
+    Map<String, Object> parsedValue = new HashMap<>();
+    for (Map.Entry<String, Object> entry : value.entrySet()) {
+      parsedValue.put(entry.getKey(), parseValue(entry.getValue()));
+    }
```
Comment (benwarfield-usds) [inline backend/src/main/java/gov/cdc/usds/simplereport/db/model/auxiliary/RequestVariablesSerializer.java:39]: Just for fun...
```suggestion
    Map<String, Object> parsedValue = value.entrySet().stream()
        .collect(Collectors.toMap(Map.Entry::getKey, e -> parseValue(e.getValue())));
```

File: backend/src/test/resources/upload-patients [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+-----------------------------23343861241310289970349754859
+Content-Disposition: form-data; name="map"
+
+{"1":["variables.patientList"]}
+-----------------------------23343861241310289970349754859
+Content-Disposition: form-data; name="1"; filename="test-upload.csv"
+Content-Type: text/csv
+
+LastName,FirstName,MiddleName,Suffix,Race,DOB,biologicalSex,Ethnicity,Street,Street2,City ,County,State,ZipCode,PhoneNumber,employedInHealthcare,residentCongregateSetting,Role,Email,facilityId
+Best,Tim,,,White,5/11/1933,Male,Not_Hispanic,123 Main Street,,Washington,,DC,20008,5656667777,Yes,No,Staff,foo@example.com,
```
Comment (benwarfield-usds) [inline backend/src/test/resources/upload-patients:14]: This doesn't exactly stress-test the upload, but thanks for getting a working example into the source tree!

File: backend/src/main/java/gov/cdc/usds/simplereport/db/model/auxiliary/RequestVariablesSerializer.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  private Object parseValue(Object value) {
+    if (value instanceof Part) {
+      Part part = (Part) value;
+      try {
+        return new BufferedReader(
+                new InputStreamReader(part.getInputStream(), StandardCharset.UTF_8))
+            .lines()
+            .collect(Collectors.joining("\n"));
+      } catch (IOException e) {
+        return "Unable to read file";
```
Comment (jeskew-gov) [inline backend/src/main/java/gov/cdc/usds/simplereport/db/model/auxiliary/RequestVariablesSerializer.java:26]: Log this and then throw a different (runtime) exception. If we can't read the file, then audit logging has failed, and we need to bail on the whole request.

File: backend/src/test/java/gov/cdc/usds/simplereport/api/PatientUploadTest.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+public class PatientUploadTest extends BaseApiTest {
+  public static final int PATIENT_PAGEOFFSET = 0;
+  public static final int PATIENT_PAGESIZE = 1000;
+
+  @Test
+  void uploadPatientCSV() throws Exception {
+    useSuperUser();
+    JsonNode response = uploadPatients();
+    assertEquals(
+        "\"Successfully uploaded 1 record(s)\"", response.get("uploadPatients").toString());
```
Comment (jeskew-gov) [inline backend/src/test/java/gov/cdc/usds/simplereport/api/PatientUploadTest.java:22]: There is an `assertLastAuditEntry` if you want to validate what was saved in the audit log, but that is a lot of copying and pasting.

**Review by jeskew-gov — APPROVED** [review]

**Pull request merged**